pkg/web/user.go

```diff
@@ -17,13 +17,47 @@
 package web
 
 // AuthenticationType specifies the authentication type that is stored in the user context
-type AuthenticationType string
+type AuthenticationType int
+
+var authenticationTypes = []string{"Basic", "Bearer"}
 
 const (
-	Basic  AuthenticationType = "Basic"
-	Bearer AuthenticationType = "Bearer"
+	Basic AuthenticationType = iota
+	Bearer
 )
 
+// String implements Stringer and converts the decision to human-readable value
+func (a AuthenticationType) String() string {
+	return authenticationTypes[a]
+}
+
+// AccessLevel specifies the access level privileges that are stored in the user context
+type AccessLevel int
+
+var levels = []string{"DefaultAccess", "GlobalAccess", "SingleTenantAccess", "AllTenantAccess"}
+
+const (
+	//DefaultAccess is the default value for access level - it is used when a component does not expliciting set an access level
+	DefaultAccess AccessLevel = iota
+
+	// GlobalAccess means access was granted to manage global resources (such resources are not scoped or associated with tenant.
+	// Such access might be granted to systems that need to manage global resources	GlobalAccess AccessLevel = iota
+	GlobalAccess
+
+	// TenantAccess means access was granted to manage the tenant's own resources. Such access might be granted to a user
+	// so the he can manage his own data
+	TenantAccess
+
+	// AllTenantAccess means access was granted to manage the resources of all tenants. Such access might be granted
+	// to systems that have to manage data across multiple tenants
+	AllTenantAccess
```

Comment:
how do `AllTenantAccess` and `GlobalAccess` relate? Do you get `AllTenantAccess` if you have `GlobalAccess`? If not then both of these give you so to say `FullAccess`, so does it make sense to allow multiple access levels in a user context?

Reply:
AllTenantAccess  allows to see and manage only tenant scoped resources (but for all tenants)

GlobalAccess allows to manage all resources (listing with global access will return both tenant and non-tenant resources). Basically GlobalAccess means a tenantID should not be  looked for at all (there will be no tenant id in criterias)